google/cloud/pubsub/publisher_connection_test.cc

```diff
@@ -363,13 +364,20 @@ TEST(MakePublisherConnectionTest, TracingEnabled) {
   auto response =
       publisher->Publish({MessageBuilder{}.SetData("test-data-0").Build()})
           .get();
+  publisher->Flush({});
 
+  auto spans = span_catcher->GetSpans();
+  ASSERT_THAT(spans, SizeIs(8));
```

Comment:
These two tests cannot both pass.  The first one asks for 8 elements, the second for 7 elements.  One of them is going to fail.  

First, I would remove the `SizeIs()` test because it is implicitly in the second test.